src/main/java/plugins/perrine/easyclemv0/sequence/SequenceUpdater.java

```diff
@@ -27,7 +27,7 @@
     private TransformationFactory transformationFactory;
     private DatasetFactory datasetFactory;
     private RoiUpdater roiUpdater;
-
+    private Transformation transformation;
```

Comment:
No, `Transformation` should not be a field of `SequenceUpdater`. Use `TransformationFactory` directly inside of `XmlTransformationWriter` injected as a dependency. Eventually cache the computed transformations in `TransformationFactory` if you want to save computation time.